src/tribler-core/tribler_core/modules/metadata_store/community/remote_query_community.py

```diff
@@ -84,11 +87,24 @@ def __init__(self, my_peer, endpoint, network, metadata_store, settings=None, no
         self.queried_subscribed_channels_peers = set()
         self.queried_peers_limit = 1000
 
+        if self.notifier:
+            self.notifier.add_observer(NTFY.POPULARITY_COMMUNITY_ADD_UNKNOWN_TORRENT,
+                                       self.on_pc_add_unknown_torrent)
+
         self.add_message_handler(RemoteSelectPayload, self.on_remote_select)
         self.add_message_handler(SelectResponsePayload, self.on_remote_select_response)
 
         self.request_cache = RequestCache()
 
+    def __del__(self):
```

Comment:
`__del__` is unreliable in Python. It will only be called during garbage cleaning of the object. Python is not C unfortunately... We use a convention to add an explicit `shutdown` method to guarantee proper object deconstruction.

Reply:
Thanks.

As I see the proper place to shutdown functionality for the `Community` class is 
```
async def unload(self)
```
Am I right?

Reply:
We use `unload` for communities and `shutdown` for everything else I guess.

Reply:
Fixed.